app/models/user.rb

```diff
@@ -10,8 +10,10 @@ class User < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   validates :email, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP }
   has_secure_password
-  
+
   after_destroy :ensure_an_admin_remains
+  before_destroy :email_verification_check_for_destroy
```

Comment:
Create an association of orders with dependent destroy and let me know the different behaviour in following scenarios.

1. Association declared above `before_destroy` callbacks.
2. Association declared below `before_destroy` callbacks.

Also find a way to check before destroy first irrespective of the ordering of the associations.

Reply:
1. Association declared below the `before_destroy` callback - Callback is called before deleting the associated objects 
2. Association declared above `before_destroy` callbacks - Callback is called after deleting the associated objects

To make sure `before_destroy` callback runs first every time, pass `prepend: true` in the `before_destroy` callback